### DISCUSSION TITLE
Why use PowerShell instead python

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [axmolengine/axmol#1309] [Q&A] [answered]
Hi,

Since we need Python in dev instructions, why not use Python scripts for build/setup instead of PowerShell scripts, that force people install more things.

If all scripts is in Python, it work on all platforms and don't need install more one tool.

Thanks.

**Comment by halx99** [comment, marked as answer]
In the future, will only powershell

why?

- powershell is more powerful for ci/cd
- powershell syntax is c-like, make programer life relex
- powershell can invoke system shell commands as-is, i.g. linux: `sudo apt install xxx`, windows: `where.exe pwsh.exe` 
- powershell is .net based and it's also can invoke c/c++ dll module
- powershell is microsoft offical supported product: https://github.com/PowerShell/PowerShell
- powershell code more simple, i.e. use 56LOC implement sync_folder, python 100+ LOC, refer https://github.com/axmolengine/axmol/blob/dev/1k/sync_folder.ps1 and implement axmol new project with 100LOC instead python 600+LOC
- powershell7 also cross platform, and on windows axmol only require system(win10) installed powershell5
